state.go

```diff
@@ -360,49 +374,51 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		}
 	}
 
-	// Also make an attempt to contact the node directly over TCP. This
-	// helps prevent confused clients who get isolated from UDP traffic
-	// but can still speak TCP (which also means they can possibly report
-	// misinformation to other nodes via anti-entropy), avoiding flapping in
-	// the cluster.
-	//
-	// This is a little unusual because we will attempt a TCP ping to any
-	// member who understands version 3 of the protocol, regardless of
-	// which protocol version we are speaking. That's why we've included a
-	// config option to turn this off if desired.
-	fallbackCh := make(chan bool, 1)
-	if (!m.config.DisableTcpPings) && (node.PMax >= 3) {
-		go func() {
-			defer close(fallbackCh)
-			didContact, err := m.sendPingAndWaitForAck(node.Address(), ping, deadline)
-			if err != nil {
-				m.logger.Printf("[ERR] memberlist: Failed fallback ping: %s", err)
-			} else {
-				fallbackCh <- didContact
-			}
-		}()
-	} else {
-		close(fallbackCh)
-	}
+	if m.transport.ProtocolType() == ProtocolTypeUDP {
```

Comment:
Ugh. So my earlier suggestion about piggybacking on the error in `WriteTo` falls apart at this line.

Reply:
Yeah, but my `net.OpError` checking as well. This is why I took that out again. It is just an optimization. Since TCP failed before, it will likely fail again. I am fine with that, considering how broken it is atm.